lib/engine/game/g_18_eus/game.rb

```diff
@@ -333,6 +334,22 @@ def apply_subsidy(corporation)
         def remove_subsidy(hex_id)
           hex_by_id(hex_id).tile.icons.reject! { |icon| icon.name.include?('subsidy') }
         end
+
+        def setup_privates
+          @companies.sort_by! { rand }
+          privates = @companies.group_by { |p| p.id[0] }
+          privates.each do | group, comps |
+            first = comps.find { |c| c.id == group + "0" }
+            comps.delete(first)
```

Comment:
comps.rotate!(comps.index { |c| c.id == "{group}0"})

Reply:
Rotate doesn't do exactly the same thing, but since it's a random order, I don't think it matters that it's a different random order.

Reply:
Right, I wasn't intending to perserve order only that its random